e2e/sync_test.go

```diff
@@ -1793,3 +1795,43 @@ func TestIssue435(t *testing.T) {
 		assertError(t, err, errS3NoSuchKey)
 	}
 }
+
+// If source path contains a special file it should not be synced
+func TestSyncSocketDestinationEmpty(t *testing.T) {
+	if runtime.GOOS == "windows" {
+		t.Skip()
+	}
+
+	t.Parallel()
+
+	s3client, s5cmd := setup(t)
+	bucket := s3BucketFromTestName(t)
+	createBucket(t, s3client, bucket)
+
+	workdir := fs.NewDir(t, t.Name())
+	defer workdir.Remove()
+
+	sockaddr := workdir.Path() + "/sock"
+	ln, err := net.Listen("unix", sockaddr)
+	if err != nil {
+		t.Log(err)
+	}
```

Comment:
Test should be aborted if socket creation fails, nothing to copy if file does not exist.

Reply:
Resolved. https://github.com/peak/s5cmd/pull/618/commits/f295f56432585308a51328d0986ed4993d7292ca